wafer/schedule/models.py

```diff
@@ -12,17 +12,60 @@
 from wafer.talks.models import Talk
 
 
-@python_2_unicode_compatible
-class Day(models.Model):
-    """Days on which the conference will be held."""
-    date = models.DateField(null=True, blank=True)
+# Validation Helpers
+def includes(obj1, obj2):
+    """Test if the times for obj1 are completely included in
+       the times for obj2"""
+    if (obj2.end_time >= obj1.end_time and
+        obj2.get_start_time() <= obj1.get_start_time()):
+        return True
+    return False
+
+def overlap(obj1, obj2):
+    """Test if obj1 and obj2 overlap on times"""
+    # obj1 and obj2 are either both ScheduleBlocks or both slots
+    if (obj2.get_start_time() >= obj1.get_start_time() and
+                    obj2.get_start_time() < obj1.end_time):
+        return True
+    elif (obj2.end_time <= obj1.end_time and
+                    obj2.end_time > obj1.get_start_time()):
+        return True
+    return includes(obj1, obj2)
 
-    def __str__(self):
-        return u'%s' % self.date.strftime('%b %d (%a)')
+
+@python_2_unicode_compatible
+class ScheduleBlock(models.Model):
+    """Blocks into which we'll break the schedule.
+    
+       Typically days, but can be shorter or longer depending on use
+       case."""
+    start_time = models.DateTimeField(null=True, blank=True)
+    end_time = models.DateTimeField(null=True, blank=True)
 
     class Meta:
-        ordering = ['date']
+        ordering = ['start_time']
 
+    def get_start_time(self):
+        """Helper method to ease validation checks."""
+        return self.start_time
+
+    def __str__(self):
+        if self.start_time.date() != self.end_time.date():
+            return u'%s - %s' % (self.start_time.strftime('%b %d (%a) %H:%M'),
+                                 self.end_time.strftime('%b %d (%a) %H:%M'))
```

Comment:
@stefanor I'm wondering how DebConf would like this rendered? If one declares that a day ends at 2am the following morning, displaying full datetimes here seems clunky, but I'm not sure what is less clunky and how to handle more complicated cases.

@drnlm If people put multiple blocks all within the same day, the date() == date() check will succeed and they'll all end up having the same day.

Maybe both of these are fine for now and we should just land this and see what to do next though.

Reply:
That's true, but I'm not sure how much effort we should put into preventing that sort of thing. I suspect we'll ultimately need to add some sort "name" field to blocks to override the default in the cases were it doesn't match what the organisers want.

Reply:
I think adding some sort of custom field to override the default is a decent option, but I'll do that as a separate PR so it's easier to sort out the details